### PR TITLE
#4736 Shabondi nodeNames definition use reference setting

### PR DESCRIPTION
**Pull request by vitojeng**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4736)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-common/src/main/java/oharastream/ohara/common/setting/SettingDef.java [hunk 1]
```diff
@@ -70,6 +70,7 @@ public static SettingDef of(String json) {
     ZOOKEEPER_CLUSTER,
     BROKER_CLUSTER,
     WORKER_CLUSTER,
+    WORKSPACE_NODES,
```
Comment (chia7712) [inline ohara-common/src/main/java/oharastream/ohara/common/setting/SettingDef.java:73]: 字面上來看，是指說該value應該從workspace裡面的節點來選擇?

Comment (chia7712) [reply ohara-common/src/main/java/oharastream/ohara/common/setting/SettingDef.java:73]: 如果我猜得沒錯的話，那是否stream, zk, bk and wk的nodeNames都應該套用? 

那如果都應該套用的話，那這個ref有跟沒有的效果是差異在哪?

Comment (vitojeng) [reply ohara-common/src/main/java/oharastream/ohara/common/setting/SettingDef.java:73]: > 字面上來看，是指說該value應該從workspace裡面的節點來選擇?

yes. 原本想法是限制在 workspace nodes 的範圍,
但由於 API 本身沒有 workspace 的定義與概念,
因此將調整為 `NODE`

**Comment by vitojeng** [comment]
@chia7712 Please take a look.

File: ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala [hunk 2]
```diff
@@ -130,8 +130,9 @@ object ShabondiDefinitions {
     .key("nodeNames")
     .orderInGroup(orderInGroup())
     .optional(Type.ARRAY)
+    .reference(SettingDef.Reference.NODE)
```
Comment (chia7712) [inline ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala:133]: Could you apply this new reference to zk, bk, wk and stream? Or you want to file another PR to address it?

Comment (vitojeng) [reply ohara-shabondi/src/main/scala/oharastream/ohara/shabondi/ShabondiDefinitions.scala:133]: Sure. Will address in another PR.

**Comment by vitojeng** [comment]
not related error

**Pull request merged**